### PR TITLE
Storio sqlite annotation processor

### LINKED ISSUE #288: Annotation Processor for generating Operations Resolvers for StorIOSQLite

### PR DESCRIPTION
**Pull request by artem-zinnatullin**
Closes #288 

Code generation!

@nikitin-da please review carefully, ask everything you want to ask and as addition, think about generated classes names, may be we should add "StorIOSQLite" prefix to avoid conflicts with generated code for "StorIOContentResolver"

### CODE REVIEW

File: storio-sqlite-annotation-processor/src/main/java/com/pushtorefresh/storio/sqlite/processor/StorIOSQLiteProcessor.java [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+        return supportedAnnotations;
+    }
+
+    @Override
+    public SourceVersion getSupportedSourceVersion() {
+        return SourceVersion.latestSupported();
+    }
+
+    /**
+     * For those who don't familiar with Annotation Processing API — this is the main method of Annotation Processor lifecycle
```
Comment (nikitin-da) [inline storio-sqlite-annotation-processor/src/main/java/com/pushtorefresh/storio/sqlite/processor/StorIOSQLiteProcessor.java]: Great doc=)

File: storio-sample-app/build.gradle [hunk 2]
```diff
@@ -44,6 +44,11 @@ dependencies {
     compile project(':storio-sqlite')
     compile project(':storio-content-resolver')
 
+    //compile files('../storio-sqlite-annotation/build/libs/storio-sqlite-annotation.jar')
+    //apt files('../storio-sqlite-annotation-processor/build/libs/storio-sqlite-annotation-processor-with-dependencies.jar')
```
Comment (artem-zinnatullin) [inline storio-sample-app/build.gradle]: I should remove commented lines

File: storio-sqlite-annotation-processor/build.gradle [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    testCompile rootProject.ext.mockitoCore
+}
+
+task createJarWithDependencies(type: Jar) {
+    baseName = project.name + '-with-dependencies'
+    from { configurations.compile.collect { it.isDirectory() ? it : zipTree(it) } }
+    with jar
+}
+
+build.dependsOn createJarWithDependencies
```
Comment (artem-zinnatullin) [inline storio-sqlite-annotation-processor/build.gradle]: Not required

File: storio-sqlite-annotation-processor/build.gradle [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    compile project(':storio-sqlite-annotation')
+    compile rootProject.ext.intellijAnnotations
+    compile rootProject.ext.autoService
+    compile rootProject.ext.javaPoet
+
+    testCompile rootProject.ext.junit
+    testCompile rootProject.ext.mockitoCore
+}
+
+task createJarWithDependencies(type: Jar) {
```
Comment (artem-zinnatullin) [inline storio-sqlite-annotation-processor/build.gradle]: Not required

File: storio-sqlite-annotation-processor/src/main/java/com/pushtorefresh/storio/sqlite/processor/StorIOSQLiteProcessor.java [hunk 5]
```diff
@@ -0,0 +1,331 @@
...
+                        + " annotation should be placed in class marked by "
+                        + StorIOSQLiteType.class.getSimpleName()
+                        + " annotation"
+                );
+            }
+
+            // Put meta column info
+            // If class already contains column with same name -> throw exception
+            if (storIOSQLiteTypeMeta.columns.put(storIOSQLiteColumnMeta.storIOSQLiteColumn.name(), storIOSQLiteColumnMeta) != null) {
+                throw new ProcessingException(annotatedFieldElement, "Column name already used in this class");
```
Comment (nikitin-da) [inline storio-sqlite-annotation-processor/src/main/java/com/pushtorefresh/storio/sqlite/processor/StorIOSQLiteProcessor.java]: What will be if we will use two different classes with the same table name?

Comment (nikitin-da) [inline storio-sqlite-annotation-processor/src/main/java/com/pushtorefresh/storio/sqlite/processor/StorIOSQLiteProcessor.java]: Oh, user may use it like feature.

Comment (artem-zinnatullin) [inline storio-sqlite-annotation-processor/src/main/java/com/pushtorefresh/storio/sqlite/processor/StorIOSQLiteProcessor.java]: Yep, I think it's okay to allow that

**Comment by nikitin-da** [comment]
It is really awesome!!! Great work man!

**Comment by artem-zinnatullin** [comment]
Thanks :)

**Pull request merged**